### PR TITLE
(test) O3-2073: Improve test coverage for the `DeathInfoSection` component

### PR DESCRIPTION
**Pull request by ayush-AI**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide) and [**design documentation**](https://zeroheight.com/23a080e38/p/880723-introduction).
- [ ] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary
I have updated the tests for the death-info-section component in this PR. I noticed that the tests were failing due to the missing isDead context, causing them to be skipped. I fixed this issue and the tests now pass successfully.

## Screenshots

*None.*

## Related Issue

https://issues.openmrs.org/browse/O3-2073

## Other

*None.*

### CODE REVIEW

**Comment by ayush-AI** [comment]
cc / @jayasanka-sack @anjula-sack @Piumal1999

File: packages/esm-patient-registration-app/src/patient-registration/section/death-info/death-info-section.test.tsx [hunk 1]
```diff
@@ -15,16 +16,18 @@ jest.mock('@openmrs/esm-framework', () => {
...
   const formValues: FormValues = initialFormValues;
 
   const setupSection = async (isDead?: boolean) => {
     render(
-      <Formik initialValues={{ ...initialFormValues, isDead }} onSubmit={null}>
-        <Form>
-          <DeathInfoSection />
-        </Form>
-      </Formik>,
+      <PatientRegistrationContext.Provider value={{ values: { isDead: isDead || false } }}>
```
Comment (vasharma05) [inline packages/esm-patient-registration-app/src/patient-registration/section/death-info/death-info-section.test.tsx:24]: isDead || false can be replaced with !!dead

Comment (vasharma05) [reply packages/esm-patient-registration-app/src/patient-registration/section/death-info/death-info-section.test.tsx:24]: isDead is not meant for `PatientRegistrationContext.Provider`.

Comment (ayush-AI) [reply packages/esm-patient-registration-app/src/patient-registration/section/death-info/death-info-section.test.tsx:24]: > isDead is not meant for `PatientRegistrationContext.Provider`.

with !!isDead. right?

Comment (ayush-AI) [reply packages/esm-patient-registration-app/src/patient-registration/section/death-info/death-info-section.test.tsx:24]: > isDead is not meant for `PatientRegistrationContext.Provider`.

dead-info-section is not rendered in the ui and the death info is not collected from the registration form but in the code there is one component for the death-info-section and also one story open for this feature https://issues.openmrs.org/browse/O3-303. So according to the story if the isDead will be checked then only this section will appear. But currently we do not have any implementation on this but isDead is defined on the FormValues types [here is the link](https://github.com/openmrs/openmrs-esm-patient-management/blob/4f28ea5189c48f22a81eb7b2f94d2efadd336b64/packages/esm-patient-registration-app/src/patient-registration/patient-registration-types.tsx#L172). 

That's why I used it here. If something need to be changed, Please let me know.

**Comment by ayush-AI** [comment]
Hello @vasharma05 could you please review this PR in your free time.

File: packages/esm-patient-registration-app/src/patient-registration/section/death-info/death-info-section.test.tsx (hunk 1 above)
Comment (Piumal1999) [reply packages/esm-patient-registration-app/src/patient-registration/section/death-info/death-info-section.test.tsx:24]: > with !!isDead. right?

Yes @ayush-AI

**Review by anjula-sack — APPROVED** [review]
Thank you! @ayush-AI

**Pull request merged**